language-plutus-core/src/Language/PlutusCore/Size.hs

```diff
@@ -0,0 +1,37 @@
+module Language.PlutusCore.Size ( termSize
+                                ) where
+
+import           Data.Functor.Foldable
+import           Language.PlutusCore.Type
+
+kindSize :: Kind a -> Int
+kindSize = cata a where
+    a TypeF{}             = 1
+    a (KindArrowF _ k k') = 1 + k + k'
+    a SizeF{}             = 1
+
+typeSize :: Type tyname a -> Int
+typeSize = cata a where
+    a TyVarF{}             = 1
```

Comment:
(aesthetics) I have become fond of `cata \case`, unless I really need to stick a type signature on the algebra.

Reply:
I'm not sure it matters either way tbh.